blocks/app.ts

```diff
@@ -0,0 +1,143 @@
+// deno-lint-ignore-file no-explicit-any
+import { propsLoader } from "$live/blocks/propsLoader.ts";
+import { SectionModule } from "$live/blocks/section.ts";
+import { FnProps } from "$live/blocks/utils.tsx";
+import { Block, BlockModule, InstanceOf } from "$live/engine/block.ts";
+import { mapObjKeys } from "$live/engine/core/utils.ts";
+import {
+  DOC_CACHE_FILE_NAME,
+  hydrateDocCacheWith,
+} from "$live/engine/schema/docCache.ts";
+import { DecoManifest, FnContext } from "$live/types.ts";
+import { once, SyncOnce } from "$live/utils/sync.ts";
+import { fromFileUrl } from "std/path/mod.ts";
+
+export type Apps = InstanceOf<typeof appBlock, "#/root/apps">;
+
+export type AppManifest = Omit<DecoManifest, "baseUrl" | "islands" | "routes">;
+
+/**
+ * @icon app-window
+ */
+export interface App<
+  TAppManifest extends AppManifest = AppManifest,
+  // deno-lint-ignore ban-types
+  TAppState = {},
+> {
+  manifest: TAppManifest;
+  state: TAppState;
+}
+
+export type AppContext<TApp extends App = App> = FnContext<
+  TApp["state"],
```

Comment:
Maybe `TApp["state"]` should merge its own state and those from its dependencies?

Something like that may be better modeled as Record<namespace, {}>.

Reply:
Discussed offline